monitoring/src/index.ts

```diff
@@ -15,53 +19,87 @@ import { WalletMonitor } from "./wallet-monitor"
+  const tbtcContracts = await loadEthereumCoreContracts(provider, chainId)
 
-const monitors: SystemEventMonitor[] = [
-  new DepositMonitor(contracts.bridge),
-  new MintingMonitor(contracts.bridge, contracts.tbtcVault, btcClient),
-  new SupplyMonitor(contracts.tbtcToken, new SupplyMonitorFilePersistence()),
-  new WalletMonitor(contracts.bridge),
-  new RedemptionMonitor(contracts.bridge),
-]
+  const bitcoinNetwork = context.bitcoinEnvironmentMapping[context.environment]
+  const btcClient = ElectrumClient.fromUrl(context.electrumUrl)
+  if ((await btcClient.getNetwork()) !== bitcoinNetwork) {
+    throw new Error("Bitcoin network mismatch")
+  }
 
-const receivers: SystemEventReceiver[] = ((): SystemEventReceiver[] => {
-  const registered: SystemEventReceiver[] = []
+  return TBTC.initializeCustom(tbtcContracts, btcClient)
+}
 
-  if (context.discordWebhookUrl) {
-    console.log("registered Discord receiver")
-    registered.push(new DiscordReceiver(context.discordWebhookUrl))
-  }
+async function setupMonitoring(sdk: TBTC): Promise<SystemEventManager> {
+  const { tbtcContracts, bitcoinClient } = sdk
 
-  if (context.sentryDsn) {
-    console.log("registered Sentry receiver")
-    registered.push(new SentryReceiver(context.sentryDsn))
-  }
+  const monitors: SystemEventMonitor[] = [
+    new DepositMonitor(tbtcContracts.bridge),
+    new MintingMonitor(
+      tbtcContracts.bridge,
+      tbtcContracts.tbtcVault,
+      bitcoinClient
+    ),
+    new SupplyMonitor(
+      tbtcContracts.tbtcToken,
+      new SupplyMonitorFilePersistence()
+    ),
+    new WalletMonitor(tbtcContracts.bridge),
+    new RedemptionMonitor(tbtcContracts.bridge),
+  ]
 
-  return registered
-})()
+  const receivers: SystemEventReceiver[] = ((): SystemEventReceiver[] => {
+    const registered: SystemEventReceiver[] = []
 
-const manager = new SystemEventManager(
-  monitors,
-  receivers,
-  new SystemEventFilePersistence()
-)
+    if (context.discordWebhookUrl) {
+      // eslint-disable-next-line no-console
+      console.log("registered Discord receiver")
+      registered.push(new DiscordReceiver(context.discordWebhookUrl))
+    }
 
-manager.trigger().then((report) => {
-  switch (report.status) {
-    case "success": {
-      console.log(report)
-      break
+    if (context.sentryDsn) {
+      // eslint-disable-next-line no-console
+      console.log("registered Sentry receiver")
+      registered.push(new SentryReceiver(context.sentryDsn))
     }
-    case "failure": {
-      console.error(report)
-      break
+
+    return registered
+  })()
+
+  return new SystemEventManager(
+    monitors,
+    receivers,
+    new SystemEventFilePersistence()
+  )
+}
+
+async function setup(): Promise<SystemEventManager> {
+  const sdk = await setupSDK()
+  return setupMonitoring(sdk)
+}
+
+setup().then((manager) => {
+  manager.trigger().then((report) => {
+    // eslint-disable-next-line default-case
```

Comment:
Couldn't we add a default case, e.g.:
```
    switch (report.status) {
      case "success": {
        console.log(report)
        break
      }
      case "failure": {
        console.error(report)
        break
      }
      default: {
        console.error("Unknown report status")
      }
    }
```
?

Reply:
This is redundant. We set the result manually to one of the two values.